src/h3/_cy/cells.pyx

```diff
@@ -281,87 +281,104 @@ cpdef int64_t num_hexagons(int resolution) except -1:
     return num_cells
 
 
-# cpdef double mean_hex_area(int resolution, unit='km^2') except -1:
-#     check_res(resolution)
+cpdef double mean_hex_area(int resolution, unit='km^2') except -1:
+    cdef:
+        h3lib.H3Error err
+        double area
 
-#     area = h3lib.hexAreaKm2(resolution)
+    check_res(resolution)
 
-#     # todo: multiple units
-#     convert = {
-#         'km^2': 1.0,
-#         'm^2': 1000*1000.0
-#     }
+    err = h3lib.getHexagonAreaAvgKm2(resolution, &area)
```

Comment:
`except -1` is unused because this doesn't propagate `err`

Reply:
`except -1` would still be used if something else failed, like `check_res`, right?

Reply:
In theory yes, but I don't think this function will return `-1`.

Reply:
This is a great discussion to have, but my plan is to defer these till later when we can examine our strategy around exceptions as a whole, instead of a function-by-function basis.

That's why I want to minimize the changes we're making at this stage. I just want us to focus on removing `h3fake2` right now. That should be a fairly straightforward and mechanical process.

Error handling/naming/API will probably require more of an open discussion, so I'm hoping to handle those as their own separate issues/PRs.

Reply:
To be discussed in #243